java-client/session-dagger/src/test/java/io/deephaven/client/AnnotatedTables.java

```diff
@@ -0,0 +1,271 @@
+
+    public static final AnnotatedTable VIEW_STATIC = ofStatic(AnnotatedTables::viewStatic, 1);
+    public static final AnnotatedTable VIEW = ofDynamic(AnnotatedTables::view);
+
+    public static final AnnotatedTable UPDATE_VIEW_STATIC = ofStatic(AnnotatedTables::updateViewStatic, 1);
+    public static final AnnotatedTable UPDATE_VIEW = ofDynamic(AnnotatedTables::updateView);
+
+    public static final AnnotatedTable UPDATE_STATIC = ofStatic(AnnotatedTables::updateStatic, 1);
+    public static final AnnotatedTable UPDATE = ofDynamic(AnnotatedTables::update);
+
+    public static final AnnotatedTable SELECT_STATIC = ofStatic(AnnotatedTables::selectStatic, 1);
+    public static final AnnotatedTable SELECT = ofDynamic(AnnotatedTables::select);
+
+    public static final AnnotatedTable HEAD_STATIC = ofStatic(AnnotatedTables::headStatic, 1);
+    public static final AnnotatedTable HEAD = ofDynamic(AnnotatedTables::head);
+
+    public static final AnnotatedTable TAIL_STATIC = ofStatic(AnnotatedTables::tailStatic, 1);
+    public static final AnnotatedTable TAIL = ofDynamic(AnnotatedTables::tail);
+
+    public static final Adapter REVERSE_ADAPTER = AnnotatedTables::reverse;
+
+    public static final Adapter HEAD_0_ADAPTER = in -> head(in, 0);
+
+    public static final Adapter HEAD_1_ADAPTER = in -> head(in, 1);
+
+    public static final Adapter HEAD_2_ADAPTER = in -> head(in, 2);
+
+    public static final Adapter TAIL_0_ADAPTER = in -> tail(in, 0);
+
+    public static final Adapter TAIL_1_ADAPTER = in -> tail(in, 1);
+
+    public static final Adapter TAIL_2_ADAPTER = in -> tail(in, 2);
+
+    public static final Adapter BY_ADAPTER = AnnotatedTables::by;
+
+    public static final Adapter MERGE_2 = in -> {
+        final TableCreationLogic merge2 = new TableCreationLogic() {
+            @Override
+            public <T extends TableOperations<T, T>> T create(TableCreator<T> creation) {
+                final T t = in.logic().create(creation);
+                return creation.merge(t, t);
+            }
+        };
+        if (in.isStatic()) {
+            return AnnotatedTable.ofStatic(merge2, in.size() * 2);
+        } else {
+            return AnnotatedTable.ofDynamic(merge2);
+        }
+    };
+
+    public static final Adapter MERGE_3 = in -> {
+        final TableCreationLogic merge3 = new TableCreationLogic() {
+            @Override
+            public <T extends TableOperations<T, T>> T create(TableCreator<T> creation) {
+                final T t = in.logic().create(creation);
+                return creation.merge(t, t, t);
+            }
+        };
+        if (in.isStatic()) {
+            return AnnotatedTable.ofStatic(merge3, in.size() * 3);
+        } else {
+            return AnnotatedTable.ofDynamic(merge3);
+        }
+    };
+
+    public static Map<String, AnnotatedTable> explicitlyAnnotatedTables() {
+        // I wish there were any easy way to get variable names or lambda function names from java :(
```

Comment:
You could use java reflection pretty easily to achieve this; but, the below table is just fine, too.